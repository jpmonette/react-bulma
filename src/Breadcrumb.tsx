import * as React from "react";
import classNames from "classnames";

interface ItemProps {
  className?: string;
  href?: string;
  isActive?: boolean;
  // TODO: add icons
}

class BreadcrumbItem extends React.Component<ItemProps> {
  render() {
    const { children, href, isActive, className, ...props } = this.props;
    return (
      <li {...props} className={classNames(className, isActive && "is-active")}>
        <a href={href} aria-current={isActive ? "page" : undefined}>
          {children}
        </a>
      </li>
    );
  }
}

interface Props {
  className?: string;
  children?: React.ReactNode;
  hasArrowSeperator?: boolean;
  hasBulletSeperator?: boolean;
  hasDotSeperator?: boolean;
  hasSucceedsSeparator?: boolean;
  isCentered?: boolean;
  isLarge?: boolean;
  isMedium?: boolean;
  isRight?: boolean;
  isSmall?: boolean;
}

class Breadcrumb extends React.Component<Props> {
  static Item = BreadcrumbItem;
  render() {
    const {
      children,
      className,
      hasArrowSeperator,
      hasBulletSeperator,
      hasDotSeperator,
      hasSucceedsSeparator,
      isCentered,
      isLarge,
      isMedium,
      isRight,
      isSmall,
      ...props
    } = this.props;

    return (
      <nav
        {...props}
        className={classNames(
          "breadcrumb",
          className,
          hasArrowSeperator && "has-arrow-seperator",
          hasBulletSeperator && "has-bullet-seperator",
          hasDotSeperator && "has-dot-seperator",
          hasSucceedsSeparator && "has-succeeds-seperator",
          isCentered && "is-centered",
          isLarge && "is-large",
          isMedium && "is-medium",
          isRight && "is-right",
          isSmall && "is-small",
        )}
        aria-label="breadcrumbs"
      >
        {children && <ul>{children}</ul>}
      </nav>
    );
  }
}

export { Breadcrumb };
