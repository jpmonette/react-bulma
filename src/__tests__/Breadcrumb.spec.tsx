import * as React from "react";
import { Breadcrumb, BreadcrumbItem as Item } from "../Breadcrumb";
import { create } from "react-test-renderer";

describe("Breadcrumb", () => {
  it("should generate an empty breadcrumb", () => {
    const tree = create(<Breadcrumb />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should generate a breadcrumb", () => {
    const children = [
      <Item key={0} href={"/home"} isActive={true}>
        Home
      </Item>,
      <Item key={1} isActive={true}>
        About
      </Item>
    ];

    const tree = create(
      <Breadcrumb hasArrowSeperator={true} className={"custom-class"} isCentered isLarge>
        {children}
      </Breadcrumb>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Breadcrumb.Item", () => {
  it("should generate an item", () => {
    const tree = create(
      <Item key={0} href={"/home"} isActive={true}>
        Home
      </Item>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
