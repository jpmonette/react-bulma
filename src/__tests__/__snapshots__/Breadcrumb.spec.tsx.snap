// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should generate a breadcrumb 1`] = `
<nav
  aria-label="breadcrumbs"
  className="breadcrumb custom-class has-arrow-seperator is-centered is-large"
>
  <ul>
    <li
      className="is-active"
    >
      <a
        aria-current="page"
        href="/home"
      >
        Home
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        aria-current="page"
      >
        About
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb should generate an empty breadcrumb 1`] = `
<nav
  aria-label="breadcrumbs"
  className="breadcrumb"
/>
`;

exports[`Breadcrumb.Item should generate an item 1`] = `
<li
  className="is-active"
>
  <a
    aria-current="page"
    href="/home"
  >
    Home
  </a>
</li>
`;
